cli/scorecard/violations.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scorecard
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"cloud.google.com/go/storage"
+
+	tfconverter "github.com/GoogleCloudPlatform/terraform-validator/converters/google"
+	"github.com/forseti-security/config-validator/pkg/api/validator"
+	"github.com/forseti-security/config-validator/pkg/gcv"
+)
+
+// attachValidator attaches a Validator to the given config
+func attachValidator(config *ScoringConfig) error {
```

Comment:
Is there a reason we don't have a NewScoringConfig method?  Also, this seems a lot like an options mixin for how Inventory operates.  Perhaps Inventory should take a ScoringConfig as a param?

Reply:
I've moved this functionality into a NewScoringConfig method.

Theoretically, different scoring configs could be used to score the same Inventory which is why I kept them separate (the Score method takes the config as a param).